pkg/cloudprovider/providers/azure/azure_routes_test.go

```diff
@@ -0,0 +1,163 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package azure
+
+import (
+	"reflect"
+	"testing"
+
+	"k8s.io/kubernetes/pkg/cloudprovider"
+
+	"github.com/Azure/azure-sdk-for-go/arm/network"
+	"github.com/Azure/go-autorest/autorest/to"
+)
+
+func TestCreateRoute(t *testing.T) {
+	fake := newFakeRouteTablesClient()
+	cloud := &Cloud{
+		RouteTablesClient: fake,
+		Config: Config{
+			ResourceGroup:  "foo",
+			RouteTableName: "bar",
+			Location:       "location",
+		},
+	}
+	expectedTable := network.RouteTable{
+		Name:     &cloud.RouteTableName,
+		Location: &cloud.Location,
+	}
+
+	err := cloud.createRouteTable()
+	if err != nil {
+		t.Errorf("unexpected error: %v", err)
```

Comment:
Looks like it should be fatal error, no?

Reply:
... also the message could be improved to mention what action failed. Or it will be in the returned error?

Reply:
I guess? It doesn't seem to be a really terrible problem if the test returns a few more error messages.  I can add `t.FailNow()` if you really want me to...

Regarding the error message, is it really important? The test error will indicate a line number and test name which makes it pretty easy to determine what failed.

Again, I can add but I'm not sure I see the point in a unit test.

Reply:
> It doesn't seem to be a really terrible problem if the test returns a few more error messages.

Yes it's not a problem when you run just a single test or only the tests from a single module. But when I open a report from Kubernetes CI with 10000 lines of output, I'd prefer to have less error messages that aren't related to a real problem.

> Regarding the error message, is it really important? The test error will indicate a line number and test name which makes it pretty easy to determine what failed.

No, it's not really important, but it's convenient for the case when another PR was merged, you did a rebase to the latest master and now a test could have different line numbers than in the CI report. In this case, you need to run unit tests to have up-to-date error message/line numbers. When error message is unique it could be easier traced to the code.

Reply:
done.